modules/internal/src/main/scala/client/calls.scala

```diff
@@ -18,26 +18,24 @@ package freestyle.rpc
 package internal
 package client
 
-import cats.~>
-import freestyle.async.AsyncContext
+import cats.effect.Effect
 import freestyle.async.guava.implicits._
+import freestyle.async.catsEffect.implicits._
 import io.grpc.{CallOptions, Channel, MethodDescriptor}
 import io.grpc.stub.{ClientCalls, StreamObserver}
-import monix.eval.Task
+import monix.execution.Scheduler
 import monix.reactive.Observable
 import org.reactivestreams._
 
-import scala.concurrent.ExecutionContext
-
 object calls {
 
   import freestyle.rpc.internal.converters._
 
-  def unary[M[_], Req, Res](
+  def unary[F[_]: Effect, Req, Res](
```

Comment:
What requires this to be an Effect? If it's `toIO` and `to[F]` those are in `LiftIO`

Reply:
For deriving the `AsyncContext` instance with [`frees-async-cats-effect`](https://github.com/frees-io/freestyle/blob/master/modules/async/async-cats-effect/shared/src/main/scala/AsyncCatsEffect.scala#L22-L29)

Reply:
Just left the same comment 😃 
`toIO` is defined on `Task` directly and requires a `Scheduler`.
If we use `LiftIO.liftIO` instead of `IO#to` we can just use `LiftIO` indeed.

Reply:
We could change the code there to `Async` instead of `Effect`.

Reply:
Do we still need `AsyncContext` after this PR?

Reply:
@peterneyens at this time `AsyncContext` is there to support `Future` since those can't implement `Sync` but would love to get rid of it if we can support scala Futures in some other way.

Reply:
Yeah, I forgot that this was at the client side of RPC.

Reply:
@peterneyens This PR is already removing AsyncContext. These [instances](https://github.com/frees-io/freestyle-rpc/blob/master/modules/async/src/main/scala/RPCAsyncImplicits.scala#L28-L39) can be also removed.

@raulraja We are not supporting `scala.concurrent.Future` before this PR, at least on the server side.

Reply:
But we need the `AsyncContext` for the [`listenableFuture2Async`](https://github.com/frees-io/freestyle/blob/master/modules/async/async-guava/src/main/scala/AsyncGuava.scala#L29)

Reply:
If we change the `Effect` to `Async` [here](https://github.com/frees-io/freestyle/blob/master/modules/async/async-cats-effect/shared/src/main/scala/AsyncCatsEffect.scala#L23), we could probably lower the `Effect` to `Async` in this PR as well.

Reply:
Addressed in https://github.com/frees-io/freestyle/pull/516